crates/oxc_cli/src/git.rs

```diff
@@ -0,0 +1,55 @@
+use std::path::Path;
+
+use git2::Repository;
+
+pub enum GitResult {
+    NoRepo,
+    MultipleRepos,
+    UncommittedChanges,
+}
+
+pub struct Git<'a> {
+    paths: &'a Vec<Box<Path>>,
+    repos: Vec<Repository>,
+}
+
+impl<'a> Git<'a> {
+    pub fn new(paths: &'a Vec<Box<Path>>) -> Self {
+        let repos: Vec<Repository> =
+            paths.iter().filter_map(|path| Repository::discover(path).ok()).collect();
+        Self { paths, repos }
+    }
+
+    pub fn verify(&'a self) -> Result<&'a Repository, GitResult> {
+        if self.repos.is_empty() {
+            return Err(GitResult::NoRepo);
+        }
+        match self.is_same_repo() {
+            Ok(repo) => {
+                for path in self.paths {
+                    if Self::is_uncommitted(repo, path) {
+                        return Err(GitResult::UncommittedChanges);
+                    }
+                }
+                Ok(repo)
+            }
+            err => err,
+        }
```

Comment:
Is there a nicer way to express a nested `Result`? I tried `Result::flatten` but looks like it's not stable yet

Reply:
I guess we need some better error handling, it should be same approach where we used `miette` in the parser and the linter. Shall we refactor these error messages in the next PR?

Reply:
Sounds good, will follow up with PR using `miette` 👍🏼